### ISSUE TITLE
Inventory directory error in all *but* Home dir (Debian)

### ISSUE THREAD

**Issue by alexphl** [moonrepo/proto#478]
### What version?

0.35.2

### Which command?

proto regen

### Description

You can use the **Dockerfile** provided below for a minimal reproduction. Running `proto regen` inside the home directory works, but if you `cd` into *any* other directory and try again you will see the following error:

```
Error: proto::tool::invalid_dir

  ×  inventory directory has been overridden but is not an absolute path. Only absolute paths are supported.
  ```

I originally came across this issue while trying to use `moon run` btw, it doesn't just affect `proto regen`.
This issue also occurs with standalone moon installations.

### Dockerfile:
```
FROM rust:latest

# Add rustfmt and clippy components
RUN rustup component add rustfmt clippy

# Install necessary packages
RUN apt-get update && apt-get install -y \
    libpq-dev \
    ca-certificates \
    libdbus-1-dev \
    librust-alsa-sys-dev \
    libudev-dev \
    protobuf-compiler \
    pkg-config \
    apt-transport-https \
    curl \
    gnupg2 \
    software-properties-common \
    git \
    unzip \
    gzip \
    xz-utils

# Add Docker’s official GPG key
RUN install -m 0755 -d /etc/apt/keyrings
RUN curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
RUN chmod a+r /etc/apt/keyrings/docker.asc

RUN echo \
      "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
      $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
    tee /etc/apt/sources.list.d/docker.list > /dev/null

# Install Docker CLI dependencies
RUN apt-get update && apt-get install -y docker-ce-cli

# Install Moonrepo
ENV PROTO_DEBUG=true
ENV MOON_DEBUG=true
ENV PROTO_HOME="$HOME/.proto"
ENV PATH="${PROTO_HOME}/bin:${PATH}"
RUN curl -fsSL https://moonrepo.dev/install/proto.sh | bash -s -- --yes
RUN proto plugin add moon "source:https://raw.githubusercontent.com/moonrepo/moon/master/proto-plugin.toml"
RUN proto install moon
RUN proto regen
```

### Operating system?

Linux

### Architecture?

x64

**Comment by milesj** [comment]
@alexphl This looks related: https://github.com/moonrepo/proto/issues/477

Can you debug based on the comments in that thread also.

**Comment by alexphl** [comment]
@milesj hey, 0.35.3 did indeed fix the issue with `proto regen`. However, `moon run` (1.24.6) still fails with the same error and preconditions. It produces the pre-0.35.3 inventory error message, so I assume it's pinned against an older proto version?

**Comment by milesj** [comment]
@alexphl Yeah moon is pinned to a specific proto version. It won't be updated until the next moon minor version.

I'm assuming you're using rust in moon? I'd suggest removing the `rust.version` setting so that it unblocks you for now.

**Comment by alexphl** [comment]
yeah, im using rust but my entire **toochain.yml** is just:
```
# Enable Rust
rust: { }
```

No `rust.version` here, but I do have RUST_VERSION set in env.
@milesj in case this means anything, I will dump whatever I think can help debugging here, and close the issue as it's out of scope for proto.

**env** (docker)
```
HOSTNAME=60ef818c2bbd
PROTO_HOME=/.proto/
PWD=/
PROTO_DEBUG=true
HOME=/root
CARGO_HOME=/usr/local/cargo
MOON_DEBUG=true
TERM=xterm
RUSTUP_HOME=/usr/local/rustup
SHLVL=1
RUST_VERSION=1.78.0
PATH=/.proto//bin:/usr/local/cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env
```
workspace.yml:
```
$schema: 'https://moonrepo.dev/schemas/workspace.json'

projects:
  rust: '.'

vcs:
  manager: 'git'
  defaultBranch: 'master'
```

moon.yml (workspace level):
```
language: 'rust'
type: 'application'

env:
  CARGO_TERM_COLOR: 'always'
  RUST_BACKTRACE: '1'

fileGroups:
  sources:
    - './**/*.rs'
    - './**/*.env.*'
    - './**/Cargo.toml'
    - './Cargo.lock'
    - '!./target/**/*'

tasks:
  hub-gateway-run:
    command: 'cargo run --bin hub_gateway'
    inputs:
      - '@globs(sources)'
    local: true
  hub-gateway-build-release:
    command: 'cargo build --bin hub_gateway --release'
    inputs:
      - '@globs(sources)'
    outputs: [ 'target/release/hub_gateway' ]
  hub-programmer-run:
    command: 'cargo run --bin hub-programmer --release'
    inputs:
      - '@globs(sources)'
    local: true
    options:
      interactive: true
  hub-programmer-build:
    command: 'cargo build --bin hub-programmer --release'
    inputs:
      - '@globs(sources)'
    outputs: [ 'target/release/hub-programmer' ]
  integration-dev-test:
    command: 'cd ./hub_simulator && cargo test --verbose'
    inputs:
      - '@globs(sources)'
  cargo-check:
    command: 'cargo check --all-features --all-targets --all'
    inputs:
      - '@globs(sources)'
  fmt-check:
    command: 'cargo fmt --all --check'
    inputs:
      - '@globs(sources)'
  fmt-all:
    command: 'cargo fmt --all'
    inputs:
      - '@globs(sources)'
    local: true
  lint-all:
    command: 'cargo clippy --workspace --all-targets --all-features'
    inputs:
      - '@globs(sources)'
  test-all:
    command: 'cargo test --workspace'
    inputs:
      - '@globs(sources)'
```

**Comment by milesj** [comment]
Hrmm yeah, I think you may be blocked on this until I can patch moon. I guess is an alternative is to downgrade moon to a point where this wasn't happening?
